Terminal.Gui/Views/GraphView.cs

```diff
@@ -0,0 +1,1913 @@
+		{
+			if (!Visible) {
+				return;
+			}
+
+			var x = GetAxisXPosition (graph, bounds);
+			var labels = GetLabels (graph, bounds);
+			var labelThickness = GetLabelThickness (labels);
+
+			foreach (var label in labels) {
+
+				graph.Move (label.ScreenLocation.X, label.ScreenLocation.Y);
+
+				// draw the tick on the axis
+				driver.AddRune (driver.RightTee);
+
+				// and the label text
+				if (!string.IsNullOrWhiteSpace (label.Text)) {
+					graph.Move (Math.Max (0, x - labelThickness), label.ScreenLocation.Y);
+					driver.AddStr (label.Text);
+				}
+			}
+
+			// if there is a title
+			if (!string.IsNullOrWhiteSpace (Text)) {
+
+				string toRender = Text;
+
+				// if label is too long
+				if (toRender.Length > graph.Bounds.Height) {
+					toRender = toRender.Substring (0, graph.Bounds.Height);
+				}
+
+				// Draw it 1 letter at a time vertically down row 0 of the control
+				int startDrawingAtY = graph.Bounds.Height / 2 - (toRender.Length / 2);
+
+				for (int i = 0; i < toRender.Length; i++) {
+
+					graph.Move (0, startDrawingAtY + i);
+					driver.AddRune (toRender [i]);
+				}
+
+			}
+		}
+
+		private IEnumerable<AxisIncrementToRender> GetLabels (GraphView graph, Rect bounds)
+		{
+			// if no labels
+			if (Increment == 0) {
+				yield break;
+			}
+
+			int labels = 0;
+
+			int x = GetAxisXPosition (graph, bounds);
+
+			for (int i = 0; i < bounds.Height; i++) {
+
+				// what bit of the graph is supposed to go here?
+				var graphSpace = graph.ScreenToGraphSpace (x, i);
+
+				if (Math.Abs (graphSpace.Y) % Increment < graph.CellSize.Y) {
+					var toRender = new AxisIncrementToRender (Orientation, new Point (x, i), graphSpace);
+
+					// and the label (if we are due one)
+					if (ShowLabelsEvery != 0) {
+
+						// if this increment also needs a label
+						if (labels++ % ShowLabelsEvery == 0) {
+							toRender.Text = LabelGetter (toRender);
+						};
+
+						yield return toRender;
+					}
+
+				}
+
+			}
+		}
+
+		/// <summary>
+		/// Returns the X screen position of the origin (typically 0,0) of graph space.
+		/// Return value is bounded by the screen i.e. the axis is always rendered even
+		/// if the origin is offscreen.
+		/// </summary>
+		/// <param name="graph"></param>
+		/// <param name="bounds"></param>
+		private int GetAxisXPosition (GraphView graph, Rect bounds)
+		{
+			// find the origin of the graph in screen space (this allows for 'crosshair' style
+			// graphs where positive and negative numbers visible
+			var origin = graph.GraphSpaceToScreen (new PointD (0, 0));
+
+			// float the Y axis so that it accurately represents the origin of the graph
+			// but anchor it to left/right if the origin is offscreen
+			return Math.Min (Math.Max ((int)graph.MarginLeft, origin.X), bounds.Width - 1);
+		}
+	}
+
+
+	/// <summary>
+	/// A location on an axis of a <see cref="GraphView"/> that may
+	/// or may not have a label associated with it
+	/// </summary>
+	public class AxisIncrementToRender {
```

Comment:
Example of a basic class that is now exposed as `public` for all of `terminal.gui`. But it has no unit tests and it is not clear if it would ever be useful outside of GraphView. 

Please consider both making it private and adding unit tests.